ChangeSet
  1.1533 04/03/30 11:27:28 anton@samba.org +1 -0
  From: Stephen Rothwell <sfr@canb.auug.org.au>
  
  This patch just ensures that the RI bit is set really early as it seems
  it is not on iSeries Power4 machines (or maybe OS400 v5r3 does this).

  arch/ppc64/kernel/head.S
    1.60 04/03/30 11:27:08 anton@samba.org +3 -0
    From: Stephen Rothwell <sfr@canb.auug.org.au>
    
    This patch just ensures that the RI bit is set really early as it seems
    it is not on iSeries Power4 machines (or maybe OS400 v5r3 does this).

diff -Nru a/arch/ppc64/kernel/head.S b/arch/ppc64/kernel/head.S
--- a/arch/ppc64/kernel/head.S	Tue Mar 30 07:07:42 2004
+++ b/arch/ppc64/kernel/head.S	Tue Mar 30 07:07:42 2004
@@ -496,6 +496,9 @@
 	.globl SystemReset_Iseries
 SystemReset_Iseries:
 	mfspr	r13,SPRG3		/* Get paca address */
+	mfmsr	r24
+	ori	r24,r24,MSR_RI
+	mtmsrd	r24			/* RI on */
 	lhz	r24,PACAPACAINDEX(r13)	/* Get processor # */
 	cmpi	0,r24,0			/* Are we processor 0? */
 	beq	.__start_initialization_iSeries	/* Start up the first processor */
.........................................................................
# vim: syntax=diff

